micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -622,7 +622,15 @@ private <M extends Meter> M registerMeterIfNecessary(Class<M> meterClass, Meter.
         return meterClass.cast(m);
     }
 
-    private Id getMappedId(Id id) {
+    Id getMappedId(Id id) {
```

Comment:
This is the same as https://github.com/micrometer-metrics/micrometer/pull/6146/files#diff-8dca70a0fea53228353bc7f6497be0f243f2370d8301f435c121336eabc2e8dfR648-R652
but without the double registration warning.

I'm thinking if we should have a version of this that does that (depending on a boolean flag).

Reply:
The link does not jump, it's line 648-652 in the changed code.